### ISSUE TITLE
New errors with latest changes of Zeex compiler

### ISSUE THREAD

**Issue by ziggi** [Open-GTO/sa-mp-fixes#78]
After https://github.com/Zeex/pawn/issues/129 and https://github.com/Zeex/pawn/pull/190 **fixes** can't be compiled with new version of compiler:
```
sa-mp-fixes/fixes.inc(8944) : error 047: array sizes do not match, or destination array is too small
sa-mp-fixes/fixes.inc(8997) : error 047: array sizes do not match, or destination array is too small
sa-mp-fixes/fixes.inc(9045) : error 047: array sizes do not match, or destination array is too small
```

@Y-Less, do you have any ideas how to fix this with more beautiful code?
https://github.com/Open-GTO/sa-mp-fixes/blob/7b9bdf65ba1bba981b7e619dbc2d2524e30f6cf1/fixes.inc#L8928-L8949

My solution for now:
```Pawn
	stock PlayerText:FIXES_CreatePlayerTextDraw(playerid, Float:x, Float:y, _FIXES_MAYBE_CONST text[])
	{
		#if FIX_CreatePlayerTextDraw_2
			_FIXES_rtrim(text);
		#endif
			#if FIX_CreatePlayerTextDraw
				static temp[1024];
				temp[0] = 0;
				if (isnull(text)) {
					strcat(temp, FIXES_gsSpace);
				} else {
					strcat(temp, text);
				}
			#else
				#define temp text
			#endif
			_FIXES_RETURN_STRING CreatePlayerTextDraw(playerid, x, y, temp);
		#if FIX_CreatePlayerTextDraw_2
			_FIXES_untrim(text);
			return ret;
		#endif
	}
```

**Comment by Y-Less** [comment]
That seems like a bug in the new compiler code, not this code.  There is no assignment, copying, `sizeof`ing, or returning of strings in that code; merely pointer passing.  I've checked the generated code in both the original compiler and the last official release of Zeex's compiler - they both generate the correct code already.  As far as I can tell, there is no reason this should be an error.

**Comment by Y-Less** [comment]
Actually caused by https://github.com/Zeex/pawn/pull/151 not checking the target is another array.

**Comment by Y-Less** [comment]
As for a temporary solution - this was using a ternary in place of an `if`, so we can move that outside the call:

```pawn
	#if FIX_CreatePlayerTextDraw
		#define _FIXES_PASS_STRING(%0(%1,%2,%3,%4)) (isnull(text) ? %0(%1,%2,%3,FIXES_gsSpace) : %0(%1,%2,%3,%4))
	#else
		#define _FIXES_PASS_STRING(%0(%1,%2,%3,%4)) %0(%1,%2,%3,%4)
	#endif
	#if FIX_CreatePlayerTextDraw_2
		#define _FIXES_RETURN_STRING new PlayerText:ret =
	#else
		#define _FIXES_RETURN_STRING return
	#endif

	stock PlayerText:FIXES_CreatePlayerTextDraw(playerid, Float:x, Float:y, _FIXES_MAYBE_CONST text[])
	{
		#if FIX_CreatePlayerTextDraw_2
			_FIXES_rtrim(text);
		#endif
			_FIXES_RETURN_STRING _FIXES_PASS_STRING(CreatePlayerTextDraw(playerid, x, y, text));
		#if FIX_CreatePlayerTextDraw_2
			_FIXES_untrim(text);
			return ret;
		#endif
	}
```

**Comment by Y-Less** [comment]
If you want to get really clever (I don't, but just for fun), write a null-coalescing operator:

```pawn
	#if FIX_CreatePlayerTextDraw
		#define _FIXES_PASS_STRING(%0(%1,%2,%3,%4??%5)) (isnull(text) ? %0(%1,%2,%3,%5) : %0(%1,%2,%3,%4))
	#else
		#define _FIXES_PASS_STRING(%0) %0
	#endif
	#if FIX_CreatePlayerTextDraw_2
		#define _FIXES_RETURN_STRING(%0) new PlayerText:ret = _FIXES_PASS_STRING(%0)
	#else
		#define _FIXES_RETURN_STRING(%0) return _FIXES_PASS_STRING(%0)
	#endif

	stock PlayerText:FIXES_CreatePlayerTextDraw(playerid, Float:x, Float:y, _FIXES_MAYBE_CONST text[])
	{
		#if FIX_CreatePlayerTextDraw_2
			_FIXES_rtrim(text);
		#endif
			_FIXES_RETURN_STRING(CreatePlayerTextDraw(playerid, x, y, text ?? FIXES_gsSpace));
		#if FIX_CreatePlayerTextDraw_2
			_FIXES_untrim(text);
			return ret;
		#endif
	}
```

**Comment by Y-Less** [comment]
Fixed in 233d12eeadbf3b151eca2cb2bb700e21ee4f7d13
